src/sections/mod.rs

```diff
@@ -300,10 +300,13 @@ impl<'a> PsdCursor<'a> {
     pub fn read_pascal_string(&mut self) -> String {
         let len = self.read_u8();
         let data = self.read(len as u32);
-        let result = String::from_utf8(data.to_vec()).unwrap();
+        let result = String::from_utf8_lossy(data).into_owned();
+
+        if len % 2 == 0 {
```

Comment:
Tests still pass without this `len % 2 == 0`

Reply:
Good catch! I forgot the problem actually occurred in another image resource block, and the original data contains a string of odd length (13) so the size field following it was incorrectly decoded (0x0005AC00 instead of 0x000005AC):

```plain
38 42 49 4D 0B B8 0D 4F 72 69 67 69 6E 44 61 74  8BIM.¸.OriginDat
61 49 52 42 00 00 05 AC 00 00 00 10 00 00 00 01  aIRB...¬........
...
```

How about removing one byte of space (0x20) from the non-UTF-8 string in the test fixture?

Reply:
Want to add a second fixture that is dedicated to this odd length pascal string case?

This way we don't stuff too many different cases into a single fixture.

If you can easily do that then that would be preferred.

If not, we can add a test to the bottom of this file that handles this case (feel free to do both if you want).

Thanks!